discord/ui/select.py

```diff
@@ -251,77 +390,416 @@ def append_option(self, option: SelectOption) -> None:
+        If not given then one is generated for you.
+    placeholder: Optional[:class:`str`]
+        The placeholder text that is shown if nothing is selected, if any.
+    min_values: :class:`int`
+        The minimum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 0 and 25.
+    max_values: :class:`int`
+        The maximum number of items that must be chosen for this select menu.
+        Defaults to 1 and must be between 1 and 25.
+    disabled: :class:`bool`
+        Whether the select is disabled or not.
+    row: Optional[:class:`int`]
+        The relative row this select menu belongs to. A Discord component can only have 5
+        rows. By default, items are arranged automatically into those 5 rows. If you'd
+        like to control the relative positioning of the row then passing an index is advised.
+        For example, row=1 will show up before row=2. Defaults to ``None``, which is automatic
+        ordering. The row number must be between 0 and 4 (i.e. zero indexed).
+    """
+
+    def __init__(
+        self,
+        *,
+        custom_id: str = MISSING,
+        placeholder: Optional[str] = None,
+        min_values: int = 1,
+        max_values: int = 1,
+        disabled: bool = False,
+        row: Optional[int] = None,
+    ) -> None:
+        super().__init__(
+            self.type,
+            custom_id=custom_id,
+            placeholder=placeholder,
+            min_values=min_values,
+            max_values=max_values,
+            disabled=disabled,
+            row=row,
         )
 
     @property
-    def type(self) -> Literal[ComponentType.select]:
-        return self._underlying.type
+    def type(self) -> Literal[ComponentType.mentionable_select]:
+        return ComponentType.mentionable_select
 
-    def is_dispatchable(self) -> bool:
-        return True
+    @property
+    def values(self) -> List[Union[Member, User, Role]]:
+        """List[Union[:class:`discord.Role`, :class:`discord.Member`, :class:`discord.User`]]: A list of roles, members,
+        and users that have been selected by the user.
+
+        If this is presented to the user in a private message, it will only allow
+        the user to select the client or themselves. Every selected option in a private
+        message will resolve to a :class:`discord.User` regardless of intents.
+
+        If invoked in a guild, the values will always resolve to :class:`discord.Member`
+        regardless of the :attr:`discord.Intents.members` intent.
+        """
+        return super().values
+
+
+class ChannelSelect(BaseSelect[V]):
+    """Represents a UI "channel" select menu with a list of predefined options representing channels in the guild.
+    It is possible to filter the channels that are shown per type by passing the ``channel_types`` parameter.
+
+    Please note that if you use this in a direct message with a user, no channels will be displayed to the user
```

Comment:
```suggestion
    Please note that if you use this in a private message with a user, no channels will be displayed to the user
```

Might as well be consistent.